packaging/release.py

```diff
@@ -0,0 +1,1410 @@
+    body = body_template.render(**variables).strip()
+
+    message = ReleaseAnnouncement(
+        subject=subject,
+        body=body,
+    )
+
+    return message
+
+
+# endregion
+# region Templates
+
+
+RELEASE_ANNOUNCEMENT_RECIPIENTS = (
+    "ansible-announce@googlegroups.com",
+    "ansible-project@googlegroups.com",
+    "ansible-devel@googlegroups.com",
+)
+
+GITHUB_RELEASE_NOTES_TEMPLATE = """
+# Changelog
+
+See the [full changelog]({{ changelog }}) for the changes included in this release.
+
+# Release Artifacts
+
+{%- for release in releases %}
+* {{ 'Wheel: ' if release.package_type == 'bdist_wheel' else 'Source:' }} [{{ release.url|basename }}]({{ release.url }}) - {{ release.size }} bytes
+  * {{ release.digest }} ({{ release.digest_algorithm }})
+{%- endfor %}
+"""
+
+# These release templates were adapted from sivel's release announcement script.
+# See: https://gist.github.com/sivel/937bc2862a9677d8db875f3b10744d8c
+
+RELEASE_ANNOUNCEMENT_SUBJECT_TEMPLATE = """
+New release{% if rc %} candidate{% elif beta %} beta{% elif alpha %} alpha{% endif %}: {{ info.name }} {{ version }}
+"""
+
+# NOTE: Gmail will automatically wrap the plain text version when sending.
+#       There's no need to perform wrapping ahead of time for normal sentences.
+#       However, lines with special formatting should be kept short to avoid unwanted wrapping.
+RELEASE_ANNOUNCEMENT_BODY_TEMPLATE = """
+Hi all- we're happy to announce the{{ " " }}
+{%- if rc -%}
+following release candidate
+{%- elif beta -%}
+beta release of
+{%- elif alpha -%}
+alpha release of
+{%- else -%}
+general release of
+{%- endif -%}:
+
+{{ info.name }} {{ version }}
+
+
+How to get it
+-------------
+
+$ python3 -m pip install --user {{ info.name }}=={{ version }}
+
+The release artifacts can be found here:
+{% for release in info.releases %}
+# {{ ' Wheel' if release.package_type == 'bdist_wheel' else 'Source' }}: {{ release.size }} bytes
+# {{ release.digest_algorithm }}: {{ release.digest }}
+{{ release.url }}
+{%- endfor %}
+
+
+What's new
+----------
+
+{% if major %}
+This release is a major release.
+{%- else -%}
+This release is a maintenance release containing numerous bugfixes.
+{% endif %}
+The full changelog can be found here:
+
+https://github.com/{{ upstream.user }}/{{ upstream.repo }}/blob/v{{ version }}/changelogs/CHANGELOG-v{{ info.short }}.rst
+
+
+Schedule for future releases
+----------------------------
+{% if rc %}
+The release candidate will become a general availability release on {{ next_ga.strftime('%-d %B %Y') }}.
+{% elif beta %}
+Subject to the need for additional beta releases, the first release candidate is scheduled for X.
+{% elif alpha %}
+Subject to the need for additional alpha releases, the first release beta is scheduled for X.
+{% else %}
+The next release candidate is planned to be released on {{ next_rc.strftime('%-d %B %Y') }}.
+The next general availability release will be one week after.
```

Comment:
A nitpick on this line, since I wrote it, and had been too lazy to do anything about it, I always delete the newline and move it up.  Could add an extra newline too.  The unnecessary newline was probably just for line length when I did it.

Reply:
I've removed the extra line break.